src/server/debug/server/server.go

```diff
@@ -901,7 +990,12 @@ type lokiLog struct {
 	Entry  *loki.Entry
 }
 
-func (s *debugServer) queryLoki(ctx context.Context, queryStr string) ([]lokiLog, error) {
+func (s *debugServer) queryLoki(ctx context.Context, queryStr string) (retResult []lokiLog, retErr error) {
+	ctx, finishSpan := log.SpanContext(ctx, "queryLoki", zap.String("queryStr", queryStr))
+	defer func() {
```

Comment:
I noticed that in other places where we end the span, we do something like:
```go
ctx, end := log.SpanContext(ctx, "queryLoki", zap.String("queryStr", queryStr))
defer end(log.Errorp(&retErr), zap.Int("logs", len(retResult)))
```
Just curious why we do it differently here.

Reply:
finishSpan instead of end?  Because we use `end` for the end time of the query :/

Oh, and error instead of errorp?  That's because we also want to record len(retResult) on the way out, and we don't have a utility function that takes a pointer to a slice and prints the length, but we do have one for errors (since that's the more common case).